lib/src/component_declaration/component_base.dart

```diff
@@ -275,23 +275,25 @@ typedef PropsModifier(Map props);
 abstract class UiProps
     extends Object with MapViewMixin, PropsMapViewMixin, ReactPropsMixin, UbiquitousDomPropsMixin, CssClassPropsMixin
     implements Map {
-  /// Adds an arbitrary prop key-value pair.
-  void addProp(propKey, value) {
+  /// Adds an arbitrary prop key-value pair if [shouldAdd] is true, otherwise, does nothing.
+  void addProp(propKey, value, [bool shouldAdd = true]) {
+    if (!shouldAdd) return;
+
     props[propKey] = value;
   }
 
-  /// Adds a Map of arbitrary props. [props] may be null.
-  void addProps(Map propMap) {
-    if (propMap == null) {
-      return;
-    }
+  /// Adds a Map of arbitrary props if [shouldAdd] is true and [propMap] is not null.
+  ///
+  /// [props] may be null.
```

Comment:
Seems like this is leftover from the old comment, which was incorrect. This line should just be removed. since we already call out above that `[propMap]` can be null.